.github/workflows/deploy.yml

```diff
@@ -1,39 +1,68 @@
-name: Update GitHub Packages
+name: Update GitHub Packages, deploy to AWS
 
 on:
   push:
+    branches:
+      - main
     tags:
       - v*
 
+env:
+  APP_VERSION: 0.1.0
+  CHART_NAME: simulator
+
 jobs:
-  package:
-    name: Package the app
+  push:
+    name: GitHub Packages Update
     runs-on: ubuntu-latest
     if: github.event_name == 'push'
+    steps:
+      - uses: actions/checkout@v2
 
-    env:
-      GO_VERSION: "1.16.3"
+      - name: Set image tag
+        run: echo "TAG=${GITHUB_REF_NAME}-${APP_VERSION}-${GITHUB_RUN_NUMBER}" >> $GITHUB_ENV
 
-    steps:
-      - name: Checkout project
-        uses: actions/checkout@v2
+      - name: Log into registry
+        run: echo "${{ secrets.GITHUB_TOKEN }}" | docker login ghcr.io -u flapflapio --password-stdin
 
-      - name: Use go${{ env.GO_VERSION }}
-        uses: actions/setup-go@v2
-        with:
-          go-version: ^${{ env.GO_VERSION }}
-      - run: go version
+      - name: Build image
+        run: make docker
 
-      - name: Build and package
-        run: |
-          # TODO: Package up the app
+      - name: Push image
+        run: make docker-push
 
   deploy:
-    name: Update GitHub Packages
+    name: Deploy to AWS
     runs-on: ubuntu-latest
-    needs: package
-
+    needs: push
+    if: github.event_name == 'push'
     steps:
-      - name: Deploy to GitHub packages
+      - uses: actions/checkout@v2
+
+      - name: Create KUBECONFIG
```

Comment:
What is the utility and the purpose of KUBECONFIG?

Reply:
The CI will deploy to our Kubernetes cluster on AWS. In order to do so, we have create a ServiceAccount resource on the cluster. To access the cluster on behalf of the service account, we generate a kubeconfig.yml file that contains an SSL certificate identifying us to the cluster.

The kubeconfig file is a configuration file for the `kubectl` utility that can be used to control the cluster. It is also used by the `helm` tool that we are using for upgrading our cluster with the new container image that we created in the previous `push` job above.